### PR TITLE
Scale Boiler Upgrade Scheme budget for number of households in simulation

### PR DESCRIPTION
**Pull request by stephanie-chan-cnz**

### CODE REVIEW

File: simulation/model.py [hunk 1]
```diff
@@ -57,6 +58,7 @@ def __init__(
             if air_source_heat_pump_price_discount_schedule
             else None
         )
+        self.household_count = household_count
```
Comment (RyanJenkinson) [inline simulation/model.py:61]: Instead of putting this in the `__init__` should we make it a `@property` ? Isn't it just:

```
    @property
    def household_count(self):
        return len(self.space.agents)
```
Don't know how this will affect the test below, though.

File: simulation/tests/test_costs.py [hunk 2]
```diff
@@ -210,10 +215,20 @@ def test_boiler_upgrade_scheme_grant_is_zero_when_grant_cap_exceeded(
         model = model_factory(
             start_datetime=datetime.datetime(2023, 1, 1, 0, 0),
         )
-        model.boiler_upgrade_scheme_cumulative_spend_gbp = 3_000_000
+
```
Comment (RyanJenkinson) [inline simulation/tests/test_costs.py:218]: I think to make this work with the `@property` you would need to add the following line here:
```
    model.add_agents([Household()] * num_households)
```
I think you used `24_600` in the factory, so you could define `num_households=24_600` in the test here and do it this way - WDYT?

**Review by RyanJenkinson — COMMENTED** [review]
Just a comment on how we could simplify the code - let me know what you think!

File: simulation/model.py (hunk 1 above)
Comment (stephanie-chan-cnz) [reply simulation/model.py:61]: There's no need to re-derive this throughout the simulation though? It's static

File: simulation/tests/test_costs.py (hunk 2 above)
Comment (stephanie-chan-cnz) [reply simulation/tests/test_costs.py:218]: I think I'd need to update a number of tests such that the model has a non-empty set of household agents :/

**Review by RyanJenkinson — APPROVED** [review]

File: simulation/tests/test_costs.py [hunk 3]
```diff
@@ -210,10 +217,23 @@ def test_boiler_upgrade_scheme_grant_is_zero_when_grant_cap_exceeded(
...
+        num_households = random.randint(0, 5)
```
Comment (RyanJenkinson) [inline simulation/tests/test_costs.py:221]: Think we need to change this to prevent a division by zero below.
```suggestion
        num_households = random.randint(1, 5)
```

**Review by RyanJenkinson — APPROVED** [review]
